### PR TITLE
Add an external secret field to registration.

### PR DESCRIPTION
**Pull request by jsha**
This is one option to try and solve
https://github.com/ietf-wg-acme/acme/issues/170.

### CODE REVIEW

File: draft-ietf-acme-acme.md [hunk 1]
```diff
@@ -894,6 +904,7 @@ Content-Type: application/jose+json
     "url": "https://example.com/acme/new-reg"
   })
   "payload": base64url({
+    "external_secret": "XYZ123",
```
Comment (cpu) [inline draft-ietf-acme-acme.md]: On L637 `external_secret` is described as a "array of URIs". Should the example be in `[ ]` brackets? I.e.: `"external_secret": [ "XYZ123" ],` ?

Comment (cpu) [inline draft-ietf-acme-acme.md]: I might have confused myself. Perhaps the `external_secret` description on L637 copied too closely the `contact` field?

Based on the discussion from #170 it seems like this field isn't intended to be used to "contact the client" at all and is later described (L935+) more accurately as an existing authenticator that the server can use to associate the client to an external system.

Comment (jsha) [inline draft-ietf-acme-acme.md]: This was my mistake. I copied the `contact` field description but failed to edit it appropriately. Fixed now.

Comment (cpu) [inline draft-ietf-acme-acme.md]: @jsha thanks!

File: draft-ietf-acme-acme.md [hunk 2]
```diff
@@ -635,9 +635,8 @@ related to this authorization. For example, the server may wish to notify the
 client about server-initiated revocation.
 
 external_secret (optional, string):
-: An array of URIs that the server can use to contact the client for issues
-related to this authorization. For example, the server may wish to notify the
-client about server-initiated revocation.
+: A secret value that the server can use to authenticate this registration as
+belonging the same entity as an already-existing system.
```
Comment (cpu) [inline draft-ietf-acme-acme.md]: There's an additional typo here:
> belonging the same entity

`s/belonging the/belonging to the/g`

**Pull request closed without merging**